timesketch/lib/analyzers/yetiindicators.py

```diff
@@ -102,33 +107,74 @@ def run(self):
         self.get_indicators("x-regex")
 
         entities_found = set()
-
-        events = self.event_stream(query_string="*", return_fields=["message"])
         total_matches = 0
-        matching_indicators = set()
-        for event in events:
-            for _id, indicator in self.intel.items():
-                regexp = indicator["compiled_regexp"]
-                if regexp.search(event.source["message"]):
-                    total_matches += 1
-                    matching_indicators.add(indicator["id"])
-                    neighbors = self.get_neighbors(indicator["id"])
-                    self.mark_event(indicator, event, neighbors)
-                    for n in neighbors:
-                        entities_found.add("{0:s}:{1:s}".format(n["name"], n["type"]))
+        new_indicators = set()
+
+        intelligence_attribute = {"data": []}
+        existing_refs = set()
+
+        try:
+            intelligence_attribute = self.sketch.get_sketch_attributes("intelligence")
+            existing_refs = {
+                ioc["externalURI"] for ioc in intelligence_attribute["data"]
+            }
+        except ValueError:
+            print("Intelligence not set on sketch, will create from scratch.")
+
+        intelligence_items = []
+
+        for _id, indicator in self.intel.items():
+
+            query_dsl = {
+                "query": {
+                    "regexp": {"message.keyword": ".*" + indicator["pattern"] + ".*"}
+                }
+            }
+
+            events = self.event_stream(query_dsl=query_dsl, return_fields=["message"])
```

Comment:
This looks a bit brittle - you are only looking for indicator pattern in `message`. Maybe this is a tradeoff for performance? Otherwise you could search all fields, but of course, that becomes more expensive.

Reply:
How would I go about searching for all fields? IIUC, plaso puts most of the interesting stuff in `message` anyways. It might be a good idea to consider searching all fields, and making this configurable? It's pretty fast as it is right now (even over 1.4M events)

Reply:
If you don't specify a specific field, opensearch will search all fields by default. But let's keep it like this for now, but can be good to revisit in the future.